### PR TITLE
nixos/qt: multiple fixes

### PR DESCRIPTION
**Pull request by thiagokokada**
## Description of changes

- Removal of top-level `with lib`
- Allow usage of module without setting `platformTheme`, so we can set the QT_PLUGIN_PATH/QML2_IMPORT_PATH paths without theming
- Add support for kvantum and some other styles
- Add myself as maintainer

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by thiagokokada** [comment]
This PR enables a bunch of new functionality:

```
qt = {
  enable = true;
  platformName = "qt5ct";
  style = "kvantum";
}
```
Is now possible, but also:

```
qt.enable = true;
```

Is also possible. This one is important because of https://github.com/NixOS/nixpkgs/pull/236770, so almost everyone that wants to use Qt properly now will need to enable this module, even if you don't want to enable any themes.

**Comment by romildo** [comment]
> This PR enables a bunch of new functionality:
> 
> ```
> qt = {
>   enable = true;
>   platformName = "qt5ct";
>   style = "kvantum";
> }
> ```

You mean `platformTheme`, right?

File: nixos/modules/config/qt.nix [hunk 1]
```diff
@@ -1,62 +1,66 @@
...
-    else if isQtStyle then [ pkgs.libsForQt5.qtstyleplugins pkgs.qt6Packages.qt6gtk2 ]
-    else if isQt5ct then [ pkgs.libsForQt5.qt5ct pkgs.qt6Packages.qt6ct ]
-    else if isLxqt then [ pkgs.lxqt.lxqt-qtplugin pkgs.lxqt.lxqt-config ]
-    else if isKde then [ pkgs.libsForQt5.plasma-integration pkgs.libsForQt5.systemsettings ]
-    else throw "`qt.platformTheme` ${cfg.platformTheme} and `qt.style` ${cfg.style} are not compatible.";
+  platformPackages = with pkgs; {
+    gnome = [ qgnomeplatform qgnomeplatform-qt6 ];
+    gtk2 = [ libsForQt5.qtstyleplugins qt6Packages.qt6gtk2 ];
+    kde = [ libsForQt5.plasma-integration libsForQt5.systemsettings ];
+    lxqt = [ lxqt.lxqt-qtplugin lxqt-lxqt-config ];
```
Comment (romildo) [inline nixos/modules/config/qt.nix:10]:
```suggestion
    lxqt = [ lxqt.lxqt-qtplugin lxqt.lxqt-config ];
```

**Review by romildo — CHANGES_REQUESTED** [review]

File: nixos/modules/config/qt.nix (hunk 1 above)
Comment (thiagokokada) [reply nixos/modules/config/qt.nix]: Done, can you be re-review.

**Review by romildo — APPROVED** [review]

**Pull request merged**